http/server/metrics.go

```diff
@@ -36,11 +36,16 @@ func (m *metricsHTTP) report(t *tracking, r *http.Request) {
 	if m == nil || m.latency == nil {
 		return
 	}
+
 	dynAttrsOpts := metric.WithAttributes(
-		semconv.HTTPRoute(t.EndpointPattern()),
-		semconv.HTTPRequestMethodKey.String(r.Method),
-		semconv.HTTPResponseStatusCode(t.responseStatus),
+		append(
+			otelhttp.CustomMetricAttributes(r),
```

Comment:
One of the problem of having the custom attributes in the contexts is that you cannot select to what layer you want them to be applied. 

So, if you set one `key=val` at the `http.server.*`  global layer, that attribute will be set in each other layer, in to the backend. Or the other way around, since you report the metric once the request is alredy served, if you set an attribute with a plugin at the `http.client.*` backend layer, it would be set up to the `http.sever.*` . 

Also, you could set the var in a plugin that is executed only in one of the backends, when some other request is already made, so one backend would have the attribute, and another one not.

I am not sure, if we want a way to "extend the labels", we should go with this approach.